src/runtime-rs/crates/resource/src/volume/hugepage.rs

```diff
@@ -0,0 +1,224 @@
+// Copyright (c) 2019-2022 Alibaba Cloud
+// Copyright (c) 2019-2022 Ant Group
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+use std::{
+    collections::HashMap,
+    fs::File,
+    io::{BufRead, BufReader},
+};
+
+use agent::Storage;
+use anyhow::{anyhow, Context, Result};
+use byte_unit::Byte;
+use kata_sys_util::fs::get_base_name;
+use kata_types::mount::KATA_EPHEMERAL_VOLUME_TYPE;
+
+use crate::share_fs::EPHEMERAL_PATH;
+
+use super::{Volume, BIND};
+
+const PROC_MOUNTS_FILE: &str = "/proc/mounts";
+const HUGETLBFS: &str = "hugetlbfs";
+const NODEV: &str = "nodev";
+
+// container hugepage
+pub(crate) struct Hugepage {
+    storage: Option<Storage>,
+    mount: oci::Mount,
+}
+
+/// handle hugepage
+impl Hugepage {
+    pub(crate) fn new(
+        mount: &oci::Mount,
+        options_map: Option<HashMap<Byte, u64>>,
+        fs_options: Option<Vec<String>>,
```

Comment:
I think no need to wrap HashMap and Vec with Option.

Reply:
fixed